File: src/main/java/ru/arty_bikini/crm/data/enums/UserGroup.java
package ru.arty_bikini.crm.data.enums;

//права доступа
public enum UserGroup {
    ADMIN(true, true, true, true, true, true, true),
    TANYA(true, true, true, true, true, true, true),
    MANAGER(false, true, true, false, true, true, true),
    WORKER(false, false, true, false, false, false, false),
    ;

    public final boolean canEditUsers;//видит таблицу пользователей
    public final boolean canViewLeads;//видит таблицу лидов,может получить всех тренеров,изменить тренара, добавить тренера
    public final boolean canViewClients;//видит таблицу клиентов
    public final boolean canViewArchive;//видит таблицу архив
    public final boolean canAddClientsLeads;//может добавлять клиента-лида
    public final boolean canEditClients;//может изменять поля у клиента-лида
    public final boolean canAddWork;//может добавлять работу на производство и интервал,
    // изменить интервал,удалить работу,разделить интервал(добавить встречу),добавить работы на отдельную встречу,
    // удалить работу со встречи,получить список работ в интервалах, интервалов по дате

    UserGroup(boolean canEditUsers, boolean canViewLeads, boolean canViewClients, boolean canViewArchive, boolean canAddClientsLeads, boolean canEditClients, boolean canAddWork) {
        this.canEditUsers = canEditUsers;
        this.canViewLeads = canViewLeads;
        this.canViewClients = canViewClients;
        this.canViewArchive = canViewArchive;
        this.canAddClientsLeads = canAddClientsLeads;
        this.canEditClients = canEditClients;
        this.canAddWork = canAddWork;
    }
}

File: common/src/commonMain/kotlin/ru/arty_bikini/crm/dto/enums/UserGroup.kt
package ru.arty_bikini.crm.dto.enums

import kotlin.jvm.JvmField

//права доступа
enum class UserGroup
    constructor(
    @JvmField val canViewHistory: Boolean,//таня и мы
    @JvmField val canEditUsers: Boolean,//таня и мы
    @JvmField val canViewUsers: Boolean,//все
    //тренера, продукты(слитный, фитнес бикини),
    // тип страз, срочность, лямки, допы
    @JvmField val canEditDict: Boolean,
    @JvmField val canViewDict: Boolean,
    //заказ
    @JvmField val canEditOrder: Boolean,
    @JvmField val canViewOrder: Boolean,
    //получить список заказов по тренеру
    @JvmField val canViewTrainer: Boolean,
    //работы и интервалы
    @JvmField val canEditWork: Boolean,
    @JvmField val canViewWork: Boolean,
    //гугол столбики
    @JvmField val canEditColumn: Boolean,
    @JvmField val canViewColumn: Boolean,
    
    override val displayName: String,
) : HasDisplayValue {
    
    ADMIN(
        canViewHistory = true,
        canEditUsers = true,
        canViewUsers = true,
        canEditDict = true,
        canViewDict = true,
        canEditOrder = true,
        canViewOrder = true,
        canViewTrainer = true,
        canEditWork = true,
        canViewWork = true,
        canEditColumn = true,
        canViewColumn = true,
        displayName = "Админ"
    ),
    TANYA(
        canViewHistory = false,
        canEditUsers = true,
        canViewUsers = true,
        
        canEditDict = true,
        canViewDict = true,
        
        canEditOrder = true,
        canViewOrder = true,
        
        canViewTrainer = true,
        
        canEditWork = true,
        canViewWork = true,
        
        canEditColumn = false,
        canViewColumn = true,
        displayName = "Татьяна"
        ),
    MANAGER(
        canViewHistory = false,
        canEditUsers = false,
        canViewUsers = true,
    
        canEditDict = false,
        canViewDict = true,
    
        canEditOrder = true,
        canViewOrder = true,
    
        canViewTrainer = false,
    
        canEditWork = false,
        canViewWork = true,
    
        canEditColumn = false,
        canViewColumn = true,
        displayName = "Менеджер"
    ),
    SEAMSTRESS(
        canViewHistory = false,
        canEditUsers = false,
        canViewUsers = true,
    
        canEditDict = false,
        canViewDict = true,
    
        canEditOrder = false,
        canViewOrder = true,
    
        canViewTrainer = false,
    
        canEditWork = false,
        canViewWork = true,
    
        canEditColumn = false,
        canViewColumn = true,
        displayName = "Швея"
    ),
    GLUER(
        canViewHistory = false,
        canEditUsers = false,
        canViewUsers = true,
    
        canEditDict = false,
        canViewDict = true,
    
        canEditOrder = false,
        canViewOrder = true,
    
        canViewTrainer = false,
    
        canEditWork = false,
        canViewWork = true,
    
        canEditColumn = false,
        canViewColumn = true,
        displayName = "Расклейщица"
    ),
    ;
    
    
}